Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp

```diff
@@ -1342,7 +1343,9 @@ std::unique_ptr<FontPlatformData> FontCache::createFontPlatformData(const FontDe
 
     auto [syntheticBold, syntheticOblique] = computeNecessarySynthesis(font.get(), fontDescription).boldObliquePair();
 
-    return makeUnique<FontPlatformData>(font.get(), size, syntheticBold, syntheticOblique, fontDescription.orientation(), fontDescription.widthVariant(), fontDescription.textRenderingMode());
+    FontPlatformData platformData(font.get(), size, syntheticBold, syntheticOblique, fontDescription.orientation(), fontDescription.widthVariant(), fontDescription.textRenderingMode());
+
+    return makeUnique<FontPlatformData>(FontPlatformData::fontPlatformDataWithAdjustedSize(WTFMove(platformData), fontDescription));
```

Comment:
I think this is the wrong way to do it. Let's not make a whole FontPlatformData, just to throw it away and make another slightly different one. Let's calculate what the difference is, and then just pass that difference into the first constructor. (The difference is just some simple math on floats.)

Reply:
We discussed offline. The approach I suggested doesn't directly work, because we can't get the x-height without a CTFont. Even if we refactored things to do it this way, the logic to populate the x-height in FontMetrics is non-trivial. And even if we compensated for that, the `override` descriptors in `@font-face` would require more custom stuff on top, which we'd have to remember to do.... The approach taken in this patch is probably better.